frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/__tests__/WhenExpressionDecorator.spec.tsx

```diff
@@ -14,20 +21,79 @@ describe('WhenExpressionDecorator', () => {
   const props: WhenExpressionDecoratorProps = {
     width: 10,
     height: 10,
-    color: 'white',
+    status: 'Failed',
+    isPipelineRun: false,
+    isFinallyTask: false,
   };
+
+  describe('When expression decorator color', () => {
+    let clonedProps = _.cloneDeep(props);
+    let shallowWrapper;
+    it('should show the diamond decorator in grey for pipeline details page', () => {
+      shallowWrapper = shallow(<WhenExpressionDecorator {...clonedProps} />);
+      const diamondShape = shallowWrapper.find('rect');
+      expect(diamondShape.exists()).toBe(true);
+      expect(diamondShape.props().fill).toBe(greyBackgroundColor.value);
+      expect(diamondShape.props().stroke).toBe(lightBorderColor.value);
+      expect(diamondShape.props().width).toBe(props.width);
+      expect(diamondShape.props().height).toBe(props.height);
+    });
+
+    it('should show the diamond decorator in light-grey for pipeline details page', () => {
+      clonedProps = {
+        ...clonedProps,
+        status: 'Failed',
+        isPipelineRun: false,
```

Comment:
How does a Pipeline is "Failed" ? I not 100% sure, but I expect that status is null/undefined for a Pipeline in the UI?

Reply:
Changed it to `Idle`